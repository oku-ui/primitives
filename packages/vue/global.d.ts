/// <reference types="vite/client" />
import 'vitest'
import type { AxeMatchers } from 'vitest-axe/matchers'

declare module 'vitest' {
  export interface AsymmetricMatchersContaining extends AxeMatchers {}
  export interface Assertion extends AxeMatchers {}
}
