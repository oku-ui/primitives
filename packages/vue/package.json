{
  "name": "vue-packages",
  "type": "module",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "dev": "vite",
    "build": "vite build",
    "preview": "vite preview"
  },
  "dependencies": {
    "@rollup/plugin-alias": "^5.1.0"
  },
  "devDependencies": {
    "@vitejs/plugin-vue": "^5.0.3",
    "@vueuse/core": "^10.7.2",
    "filter-anything": "^3.0.7",
    "typescript": "^5.3.3",
    "unplugin-vue-components": "^0.26.0",
    "vite-plugin-pages": "^0.32.0",
    "vue-router": "^4.2.5",
    "vue-tsc": "^2.0.0"
  }
}
