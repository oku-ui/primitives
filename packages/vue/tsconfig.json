{
  "compilerOptions": {
    "target": "ESNext",
    "jsx": "preserve",
    "jsxImportSource": "vue",
    "lib": [
      "ESNext",
      "DOM",
      "DOM.Iterable"
    ],
    "useDefineForClassFields": true,
    "baseUrl": ".",
    "module": "ESNext",
    "moduleResolution": "bundler",
    "paths": {
      "@/*": [
        "src/*"
      ],
      "@oku-ui/accordion": [
        "src/accordion"
      ],
      "@oku-ui/alert-dialog": [
        "src/alert-dialog"
      ],
      "@oku-ui/arrow": [
        "src/arrow"
      ],
      "@oku-ui/aspect-ratio": [
        "src/aspect-ratio"
      ],
      "@oku-ui/avatar": [
        "src/avatar"
      ],
      "@oku-ui/collection": [
        "src/collection"
      ],
      "@oku-ui/collapsible": [
        "src/collapsible"
      ],
      "@oku-ui/checkbox": [
        "src/checkbox"
      ],
      "@oku-ui/dialog": [
        "src/dialog"
      ],
      "@oku-ui/direction": [
        "src/direction"
      ],
      "@oku-ui/dismissable-layer": [
        "src/dismissable-layer"
      ],
      "@oku-ui/focus-guards": [
        "src/focus-guards"
      ],
      "@oku-ui/focus-scope": [
        "src/focus-scope"
      ],
      "@oku-ui/hover-card": [
        "src/hover-card"
      ],
      "@oku-ui/label": [
        "src/label"
      ],
      "@oku-ui/menu": [
        "src/menu"
      ],
      "@oku-ui/popover": [
        "src/popover"
      ],
      "@oku-ui/popper": [
        "src/popper"
      ],
      "@oku-ui/portal": [
        "src/portal"
      ],
      "@oku-ui/primitive": [
        "src/primitive"
      ],
      "@oku-ui/progress": [
        "src/progress"
      ],
      "@oku-ui/provide": [
        "src/provide"
      ],
      "@oku-ui/radio-group": [
        "src/radio-group"
      ],
      "@oku-ui/roving-focus": [
        "src/roving-focus"
      ],
      "@oku-ui/scroll-area": [
        "src/scroll-area"
      ],
      "@oku-ui/separator": [
        "src/separator"
      ],
      "@oku-ui/slider": [
        "src/slider"
      ],
      "@oku-ui/slot": [
        "src/slot"
      ],
      "@oku-ui/switch": [
        "src/switch"
      ],
      "@oku-ui/tabs": [
        "src/tabs"
      ],
      "@oku-ui/toggle": [
        "src/toggle"
      ],
      "@oku-ui/toggle-group": [
        "src/toggle-group"
      ],
      "@oku-ui/toast": [
        "src/toast"
      ],
      "@oku-ui/toolbar": [
        "src/toolbar"
      ],
      "@oku-ui/tooltip": [
        "src/tooltip"
      ],
      "@oku-ui/use-composable": [
        "src/use-composable"
      ],
      "@oku-ui/utils": [
        "src/utils"
      ],
      "@oku-ui/visually-hidden": [
        "src/visually-hidden"
      ],
      "@oku-ui/presence": [
        "src/presence"
      ]
    },
    "resolveJsonModule": true,
    "allowImportingTsExtensions": true,
    "strict": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noEmit": true,
    "isolatedModules": true,
    "skipLibCheck": true
  },
  "references": [
    {
      "path": "./tsconfig.node.json"
    }
  ],
  "include": [
    "src/**/*.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "*.d.ts"
  ]
}
