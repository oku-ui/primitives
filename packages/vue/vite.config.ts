import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'
import Pages from 'vite-plugin-pages'

const resolve = (val: string) => new URL(val, import.meta.url).pathname

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    vue(),
    Pages(),
  ],
  server: {
    fs: {
      // Allow serving files from two level up to the project root
      allow: ['..'],
    },
  },
  build: {
    watch: {
      include: ['../components/**'],
    },
  },
  resolve: {
    alias: [
      // Components
      {
        find: '@oku-ui/scroll-area',
        replacement: resolve('./src/scroll-area'),
      },
      {
        find: '@oku-ui/checkbox',
        replacement: resolve('./src/checkbox'),
      },
      {
        find: '@oku-ui/alert-dialog',
        replacement: resolve('./src/alert-dialog'),
      },
      {
        find: '@oku-ui/slider',
        replacement: resolve('./src/slider'),
      },
      {
        find: '@oku-ui/label',
        replacement: resolve('./src/label'),
      },
      {
        find: '@oku-ui/separator',
        replacement: resolve('./src/separator'),
      },
      {
        find: '@oku-ui/popover',
        replacement: resolve('./src/popover'),
      },
      {
        find: '@oku-ui/toolbar',
        replacement: resolve('./src/toolbar'),
      },
      {
        find: '@oku-ui/toast',
        replacement: resolve('./src/toast'),
      },
      {
        find: '@oku-ui/toggle',
        replacement: resolve('./src/toggle'),
      },
      {
        find: '@oku-ui/toggle-group',
        replacement: resolve('./src/toggle-group'),
      },
      {
        find: '@oku-ui/switch',
        replacement: resolve('./src/switch'),
      },
      {
        find: '@oku-ui/radio-group',
        replacement: resolve('./src/radio-group'),
      },
      {
        find: '@oku-ui/avatar',
        replacement: resolve('./src/avatar'),
      },
      {
        find: '@oku-ui/hover-card',
        replacement: resolve('./src/hover-card'),
      },
      {
        find: '@oku-ui/collapsible',
        replacement: resolve('./src/collapsible'),
      },
      {
        find: '@oku-ui/progress',
        replacement: resolve('./src/progress'),
      },
      {
        find: '@oku-ui/tooltip',
        replacement: resolve('./src/tooltip'),
      },
      {
        find: '@oku-ui/aspect-ratio',
        replacement: resolve('./src/aspect-ratio'),
      },
      {
        find: '@oku-ui/tabs',
        replacement: resolve('./src/tabs'),
      },
      {
        find: '@oku-ui/dialog',
        replacement: resolve('./src/dialog'),
      },
      {
        find: '@oku-ui/primitives',
        replacement: resolve('./src/primitives'),
      },
      {
        find: '@oku-ui/accordion',
        replacement: resolve('./src/accordion'),
      },

      // Core
      {
        find: '@oku-ui/menu',
        replacement: resolve('./src/menu'),
      },
      {
        find: '@oku-ui/dismissable-layer',
        replacement: resolve('./src/dismissable-layer'),
      },
      {
        find: '@oku-ui/roving-focus',
        replacement: resolve('./src/roving-focus'),
      },
      {
        find: '@oku-ui/focus-scope',
        replacement: resolve('./src/focus-scope'),
      },
      {
        find: '@oku-ui/presence',
        replacement: resolve('./src/presence'),
      },
      {
        find: '@oku-ui/direction',
        replacement: resolve('./src/direction'),
      },
      {
        find: '@oku-ui/arrow',
        replacement: resolve('./src/arrow'),
      },
      {
        find: '@oku-ui/visually-hidden',
        replacement: resolve('./src/visually-hidden'),
      },
      {
        find: '@oku-ui/popper',
        replacement: resolve('./src/popper'),
      },
      {
        find: '@oku-ui/collection',
        replacement: resolve('./src/collection'),
      },
      {
        find: '@oku-ui/slot',
        replacement: resolve('./src/slot'),
      },
      {
        find: '@oku-ui/portal',
        replacement: resolve('./src/portal'),
      },
      {
        find: '@oku-ui/focus-guards',
        replacement: resolve('./src/focus-guards'),
      },
      {
        find: '@oku-ui/use-composable',
        replacement: resolve('./src/use-composable'),
      },
      {
        find: '@oku-ui/provide',
        replacement: resolve('./src/provide'),
      },
      {
        find: '@oku-ui/primitive',
        replacement: resolve('./src/primitive'),
      },
      {
        find: '@oku-ui/utils',
        replacement: resolve('./src/utils'),
      },
    ],
  },
})
