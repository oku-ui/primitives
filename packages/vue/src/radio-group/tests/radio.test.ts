import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import type { Component } from 'vue'
import { defineComponent, h } from 'vue'
import { OkuRadioGroup, OkuRadioGroupIndicator, OkuRadioGroupItem } from '../'

const testComponent = defineComponent({
  components: {
    OkuRadioGroup,
    OkuRadioGroupItem,
    OkuRadioGroupIndicator,
  },
  template: `
    <OkuRadioGroup default-value="1" class="root-class">

      <OkuRadioGroupItem value="1" class="item-class">
        <OkuRadioGroupIndicator class="indicator-class" />
      </OkuRadioGroupItem>

      <OkuRadioGroupItem value="2" class="item-class">
        <OkuRadioGroupIndicator class="indicator-class" />
      </OkuRadioGroupItem>

      <OkuRadioGroupItem value="3" class="item-class">
        <OkuRadioGroupIndicator class="indicator-class" />
      </OkuRadioGroupItem>

  </OkuRadioGroup>
  `,
})

const component = {
  setup(props, { attrs, slots }) {
    return () => h(OkuRadioGroup, { ...attrs }, slots)
  },
} as Component

describe('okuRadioGroup', () => {
  it('renders the component correctly', () => {
    const wrapper = mount(component)
    expect(wrapper.exists()).toBe(true)
  })

  it('renders the component correctly with a label', () => {
    const wrapper = mount(component, {
      slots: {
        default: 'Label',
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('can be checked', async () => {
    const wrapper = mount(testComponent)
    expect(wrapper.element).toMatchSnapshot()

    const value2 = wrapper.find('[value="2"]')
    await value2.trigger('click')
    expect(wrapper.element).toMatchSnapshot()

    const value3 = wrapper.find('[value="3"]')
    await value3.trigger('click')
    expect(wrapper.element).toMatchSnapshot()
    expect(value3.attributes('aria-checked')).toBe('true')
    expect(value2.attributes('aria-checked')).toBe('false')

    const span = value3.find('span')
    expect(span.classes()).toContain('indicator-class')
    expect(span.attributes('data-state')).toBe('checked')

    expect(value2.find('span').exists()).toBe(false)

    const value1 = wrapper.find('[value="1"]')
    await value1.trigger('click')
    expect(wrapper.element).toMatchSnapshot()

    expect(value1.attributes('aria-checked')).toBe('true')
    expect(value2.attributes('aria-checked')).toBe('false')
    expect(value3.attributes('aria-checked')).toBe('false')

    expect(value1.find('span').exists()).toBe(true)
    expect(value2.find('span').exists()).toBe(false)
    expect(value3.find('span').exists()).toBe(false)
  })
})
