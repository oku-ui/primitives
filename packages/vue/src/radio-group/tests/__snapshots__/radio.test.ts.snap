// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuRadioGroup > can be checked 1`] = `
<div
  data-v-app=""
>
  
  <div
    aria-required="false"
    class="root-class"
    dir="ltr"
    role="radiogroup"
    style="outline-color: none; outline-style: none; outline-width: initial;"
    tabindex="-1"
  >
    
    <button
      aria-checked="true"
      class="item-class"
      data-oku-collection-item=""
      data-state="checked"
      role="radio"
      tabindex="-1"
      type="button"
      value="1"
    >
      <span
        class="indicator-class"
        data-state="checked"
        disabled="false"
        loop="true"
        required="false"
      />
    </button>
    <!---->
    
    
    <button
      aria-checked="false"
      class="item-class"
      data-oku-collection-item=""
      data-state="unchecked"
      role="radio"
      tabindex="-1"
      type="button"
      value="2"
    >
      <!---->
    </button>
    <!---->
    
    
    <button
      aria-checked="false"
      class="item-class"
      data-oku-collection-item=""
      data-state="unchecked"
      role="radio"
      tabindex="-1"
      type="button"
      value="3"
    >
      <!---->
    </button>
    <!---->
    
  </div>
  
</div>
`;

exports[`okuRadioGroup > can be checked 2`] = `
<div
  data-v-app=""
>
  
  <div
    aria-required="false"
    class="root-class"
    dir="ltr"
    role="radiogroup"
    style="outline-color: none; outline-style: none; outline-width: initial;"
    tabindex="0"
  >
    
    <button
      aria-checked="false"
      class="item-class"
      data-oku-collection-item=""
      data-state="unchecked"
      role="radio"
      tabindex="-1"
      type="button"
      value="1"
    >
      <!---->
    </button>
    <!---->
    
    
    <button
      aria-checked="true"
      class="item-class"
      data-oku-collection-item=""
      data-state="checked"
      role="radio"
      tabindex="-1"
      type="button"
      value="2"
    >
      <span
        class="indicator-class"
        data-state="checked"
        disabled="false"
        loop="true"
        required="false"
      />
    </button>
    <!---->
    
    
    <button
      aria-checked="false"
      class="item-class"
      data-oku-collection-item=""
      data-state="unchecked"
      role="radio"
      tabindex="-1"
      type="button"
      value="3"
    >
      <!---->
    </button>
    <!---->
    
  </div>
  
</div>
`;

exports[`okuRadioGroup > can be checked 3`] = `
<div
  data-v-app=""
>
  
  <div
    aria-required="false"
    class="root-class"
    dir="ltr"
    role="radiogroup"
    style="outline-color: none; outline-style: none; outline-width: initial;"
    tabindex="0"
  >
    
    <button
      aria-checked="false"
      class="item-class"
      data-oku-collection-item=""
      data-state="unchecked"
      role="radio"
      tabindex="-1"
      type="button"
      value="1"
    >
      <!---->
    </button>
    <!---->
    
    
    <button
      aria-checked="false"
      class="item-class"
      data-oku-collection-item=""
      data-state="unchecked"
      role="radio"
      tabindex="-1"
      type="button"
      value="2"
    >
      <!---->
    </button>
    <!---->
    
    
    <button
      aria-checked="true"
      class="item-class"
      data-oku-collection-item=""
      data-state="checked"
      role="radio"
      tabindex="-1"
      type="button"
      value="3"
    >
      <span
        class="indicator-class"
        data-state="checked"
        disabled="false"
        loop="true"
        required="false"
      />
    </button>
    <!---->
    
  </div>
  
</div>
`;

exports[`okuRadioGroup > can be checked 4`] = `
<div
  data-v-app=""
>
  
  <div
    aria-required="false"
    class="root-class"
    dir="ltr"
    role="radiogroup"
    style="outline-color: none; outline-style: none; outline-width: initial;"
    tabindex="0"
  >
    
    <button
      aria-checked="true"
      class="item-class"
      data-oku-collection-item=""
      data-state="checked"
      role="radio"
      tabindex="-1"
      type="button"
      value="1"
    >
      <span
        class="indicator-class"
        data-state="checked"
        disabled="false"
        loop="true"
        required="false"
      />
    </button>
    <!---->
    
    
    <button
      aria-checked="false"
      class="item-class"
      data-oku-collection-item=""
      data-state="unchecked"
      role="radio"
      tabindex="-1"
      type="button"
      value="2"
    >
      <!---->
    </button>
    <!---->
    
    
    <button
      aria-checked="false"
      class="item-class"
      data-oku-collection-item=""
      data-state="unchecked"
      role="radio"
      tabindex="-1"
      type="button"
      value="3"
    >
      <!---->
    </button>
    <!---->
    
  </div>
  
</div>
`;

exports[`okuRadioGroup > renders the component correctly with a label 1`] = `
<div
  data-v-app=""
>
  
  <div
    aria-required="false"
    dir="ltr"
    role="radiogroup"
    style="outline-color: none; outline-style: none; outline-width: initial;"
    tabindex="-1"
  >
    Label
  </div>
  
</div>
`;
